js/apps/admin-ui/src/user/UserGroups.tsx

```diff
@@ -63,111 +62,32 @@ export const UserGroups = ({ user }: UserGroupsProps) => {
     const searchParam = search || "";
     if (searchParam) {
       params.search = searchParam;
-      setSearch(searchParam);
     }
 
     const joinedUserGroups = await adminClient.users.listGroups({
       ...params,
       id: user.id!,
     });
 
-    const allCreatedGroups = await adminClient.groups.find();
-
-    const getAllPaths = joinedUserGroups.reduce(
-      (acc: string[], cur) => (cur.path && acc.push(cur.path), acc),
-      []
-    );
-    const parentGroupNames: string[] = [];
-    const allGroupMembership: string[] = [];
-    const slicedGroups: string[] = [];
-    const rootLevelGroups: GroupRepresentation[] = [...allCreatedGroups];
-    let allPaths: GroupRepresentation[] = [];
-
-    const getAllSubgroupPaths = (
-      o: any,
-      f: any,
-      context: GroupRepresentation[]
-    ): GroupRepresentation[] => {
-      f(o, context);
-      if (typeof o !== "object") return context;
-      if (Array.isArray(o))
-        return o.forEach((e) => getAllSubgroupPaths(e, f, context)), context;
-      for (const prop in o) getAllSubgroupPaths(o[prop], f, context);
-      return context;
-    };
-
-    const arr = getAllSubgroupPaths(
-      rootLevelGroups,
-      (
-        x: GroupRepresentation | undefined,
-        context: GroupRepresentation[][]
-      ) => {
-        if (x?.subGroups) context.push(x.subGroups);
-      },
-      []
-    );
-
-    const allSubgroups: GroupRepresentation[] = [].concat(...(arr as any));
-
-    allPaths = [...rootLevelGroups, ...allSubgroups];
-
-    getAllPaths.forEach((item) => {
-      const paths = item.split("/");
-      const groups: string[] = [];
-
-      paths.reduce((acc, value) => {
-        const path = acc + "/" + value;
-        groups.push(path);
-        return path;
-      }, "");
-
-      for (let i = 1; i < groups.length; i++) {
-        slicedGroups.push(groups[i].substring(1));
-      }
-    });
-
-    allGroupMembership.push(...slicedGroups);
-
-    allPaths.forEach((item) => {
-      if (item.subGroups!.length !== 0) {
-        allPaths.push(...item!.subGroups!);
-      }
-    });
-
-    allPaths = allPaths.filter((group) =>
-      allGroupMembership.includes(group.path as any)
-    );
-
-    const topLevelGroups = allCreatedGroups.filter((value) =>
-      parentGroupNames.includes(value.name!)
-    );
-
-    const subgroupArray: any[] = [];
-
-    topLevelGroups.forEach((group) => subgroupArray.push(group.subGroups));
-
-    const directMembership = joinedUserGroups!.filter(
-      (value) => !topLevelGroups.includes(value)
-    );
-
-    setDirectMembershipList(directMembership);
-
-    const filterDupesfromGroups = allPaths.filter(
-      (thing, index, self) =>
-        index === self.findIndex((t) => t.name === thing.name)
-    );
-
-    if (!isDirectMembership) {
-      return alphabetize(filterDupesfromGroups);
-    }
+    setDirectMembershipList([...joinedUserGroups]);
+
+    if (!isDirectMembership)
+      joinedUserGroups.map((g) => {
```

Comment:
This value is mapped, but not assigned to anything, looks like this could just be a `for...of` loop?